components/ide/jetbrains/gateway-plugin/src/main/kotlin/io/gitpod/jetbrains/gateway/GitpodConnectionProvider.kt

```diff
@@ -144,6 +148,24 @@ class GitpodConnectionProvider : GatewayConnectionProvider {
             connectionPanel.repaint()
         }
 
+        val connectionKeyId = "${connectParams.gitpodHost}-${connectParams.workspaceId}-${connectParams.backendPort}"
+
+        if (activeConnections.containsKey(connectionKeyId)) {
```

Comment:
This should be thread-safe to my understanding, using `putIfAbsent`.

I've tried other approaches like `getOrPut` and `computeIfPresent` but with that approach, I didn't find a way to have an "else" condition.